include/eve/module/core/constant/constant.hpp

```diff
@@ -53,13 +52,12 @@ namespace eve
 //! @}
 //================================================================================================
 
-template<typename T, auto BitsPattern>
-requires(plain_scalar_value<element_type_t<T>>)
+template<value T, auto BitsPattern>
 EVE_FORCEINLINE auto Constant(eve::as<T> const& = {})
 {
-  using t_t = element_type_t<T>;
+  using t_t = translate_element_type_t<T>;
 
-  if constexpr( std::is_integral_v<t_t> ) { return static_cast<T>(BitsPattern); }
+  if constexpr( std::integral<t_t> ) { return static_cast<T>(BitsPattern); }
```

Comment:
I'm ok with it but I'm not sure that's better.
Concepts sometimes are slower to compile std::same_as should be slower than is_same_as_v.

Probably unimportant though

Reply:
I changed mostly for consistency